Clean up sortReverseTopological copy pasta

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44995 Clean up sortReverseTopological copy pasta**
* #44972 [JIT] Regularize tensorexpr fuser strategy with other fusers
* #44238 Refactor subgraph merging

Differential Revision: [D23821582](https://our.internmc.facebook.com/intern/diff/D23821582)

Commit:
Clean up sortReverse copy pasta

[ghstack-poisoned]

Commit:
Update on "Clean up sortReverseTopological copy pasta"




[ghstack-poisoned]

bertmaher commented on torch/csrc/jit/ir/ir.h:
```diff
@@ -1421,6 +1421,9 @@ TORCH_API std::vector<Value*> inlineCallTo(
  */
 TORCH_API std::vector<Value*> unpackOutputs(const std::vector<Value*>& outputs);
 
+// Sort inputs topologically, filtering node's who's owning block != b
```
nit: `node's` -> `nodes`

Krovatkin reviewed (approved):
:shipit:

Commit:
Update on "Clean up sortReverseTopological copy pasta"




[ghstack-poisoned]

Commit:
Update on "Clean up sortReverseTopological copy pasta"


Differential Revision: [D23821582](https://our.internmc.facebook.com/intern/diff/D23821582)

[ghstack-poisoned]

Commit:
Update on "Clean up sortReverseTopological copy pasta"


Differential Revision: [D23821582](https://our.internmc.facebook.com/intern/diff/D23821582)

[ghstack-poisoned]

Commit:
Update on "Clean up sortReverseTopological copy pasta"


Differential Revision: [D23821582](https://our.internmc.facebook.com/intern/diff/D23821582)

[ghstack-poisoned]

Commit:
Update on "Clean up sortReverseTopological copy pasta"


Differential Revision: [D23821582](https://our.internmc.facebook.com/intern/diff/D23821582)

[ghstack-poisoned]

Commit:
Update on "Clean up sortReverseTopological copy pasta"


Differential Revision: [D23821582](https://our.internmc.facebook.com/intern/diff/D23821582)

[ghstack-poisoned]

Commit:
Update on "Clean up sortReverseTopological copy pasta"


Differential Revision: [D23821582](https://our.internmc.facebook.com/intern/diff/D23821582)

[ghstack-poisoned]
